File: 61.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def rotateRight(self, head: ListNode, k: int) -> ListNode:
        if k == 0 or not head: return head
        
        length = 0
        rear = head
        while rear.next:
            length += 1
            rear = rear.next
        length += 1
        
        k = k % length
        if k == 0: return head

        right_head = head
        for i in range(length - k - 1):
            right_head = right_head.next
        tmp = right_head
        right_head = right_head.next
        tmp.next = None
        rear.next = head
        head = right_head
        return head
        
        

File: 61.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
  		if(head == NULL) return head;
  		int len = 0;
  		auto thead = head;
  		while(thead -> next) {
  			thead = thead -> next;
  			++len;
  		}
  		++len;
  		int num = k % len;
  		if(num == 0) return head;
  		int tar = len - num;
  		cout << tar << " " << num;
  		auto t = head;
  		while(--tar) {
  			t = t -> next;
  		}
  		auto newHead = t -> next;
  		t -> next = NULL;
  		thead -> next = head;
  		head = newHead;
  		return head;
    }
};